src/main/java/core/basesyntax/model/OperationModel.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.model;
+
+public class OperationModel {
+    private Operation operation;
+    private FruitModel fruit;
+    private int quantity;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public FruitModel getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(FruitModel fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation getOperationFromCode(String code) {
+            for (Operation operation : Operation.values()) {
+                if (operation.getCode().equals(code)) {
+                    return operation;
+                }
+            }
+            throw new RuntimeException("Invalid operation type " + code);
```

Comment:
```suggestion
            throw new IllegalArgumentException("Invalid operation type " + code);
```
better throw more specific exceptions where possible